lib/src/command/add.dart

```diff
@@ -285,52 +308,71 @@ class AddCommand extends PubCommand {
   ///
   /// If any of the other git options are defined when `--git-url` is not
   /// defined, an error will be thrown.
-  _ParseResult _parsePackage(String package, LanguageVersion languageVersion) {
-    final conflictingFlagSets = [
-      ['git-url', 'git-ref', 'git-path'],
-      ['hosted-url'],
-      ['path'],
-      ['sdk'],
-    ];
-
-    for (final flag
-        in conflictingFlagSets.expand((s) => s).where(argResults.wasParsed)) {
-      final conflictingFlag = conflictingFlagSets
-          .where((s) => !s.contains(flag))
-          .expand((s) => s)
-          .firstWhereOrNull(argResults.wasParsed);
-      if (conflictingFlag != null) {
-        usageException(
-            'Packages can only have one source, "pub add" flags "--$flag" and '
-            '"--$conflictingFlag" are conflicting.');
-      }
+  _ParseResult _parsePackage(
```

Comment:
Update documentation.

It would be great to have examples for both the old and new supported way.

Reply:
Done